.github/workflows/llama-perf.yml

```diff
@@ -1,140 +1,276 @@
+      fail-fast: false
+      matrix:
+        jax-version: ["0.6.0", "0.7.1"]
+        include:
+          - jax-version: "0.6.0"
+            jaxlib-version: "0.6.0"
+            docker-image: "rocm/jax:rocm7.0.2-jax0.6.0-py3.12-ubu24"
+          - jax-version: "0.7.1"
+            jaxlib-version: "0.7.1"
+            docker-image: "ghcr.io/rocm/jax-ubu24.rocm702:a2dc0f639643e3e4d03088b77114527b59c377c1" # TODO: tag nightly
+    env:
+      NVTE_FRAMEWORK: jax
+      NVTE_USE_ROCM: 1
+      HIP_PLATFORM: amd
+      NVTE_FUSED_ATTN_AOTRITON: 0
+      CU_NUM: 64
     steps:
-      - name: Checkout plugin repo
+      - name: Checkout TE source
         uses: actions/checkout@v4
         with:
-          ref: rocm-jaxlib-v0.6.0
-      - name: Download wheels
+          repository: rocm/transformerengine
+          ref: dev
+          submodules: recursive
+          fetch-depth: 0
+          path: transformerengine
+      - name: Print workspace
         run: |
-         mkdir -p wheelhouse && cd wheelhouse/
-         curl -O -L https://github.com/ROCm/rocm-jax/releases/download/rocm-jax-v0.6.0/jax_rocm7_pjrt-0.6.0-py3-none-manylinux_2_28_x86_64.whl
-         curl -O -L https://github.com/ROCm/rocm-jax/releases/download/rocm-jax-v0.6.0/jax_rocm7_plugin-0.6.0-cp312-cp312-manylinux_2_28_x86_64.whl
-      - name: Build docker
+          ls -la ${{ github.workspace }}
+      - name: Get TE commit SHA
+        id: tsha
+        run: echo "hash=$(git -C transformerengine rev-parse HEAD)" >> $GITHUB_OUTPUT
+      - name: Define ROCm arch & runner-label
+        id: arch
+        run: |
+          if [ "${{ inputs.runner-label }}" == "linux-mi325-8" ]; then
+            echo "ROCM_ARCH=gfx942" >> $GITHUB_ENV
+            echo "arch=gfx942" >> $GITHUB_OUTPUT
+            echo "label=MI325" >> $GITHUB_OUTPUT
+          else
+            echo "ROCM_ARCH=gfx950" >> $GITHUB_ENV
+            echo "arch=gfx950" >> $GITHUB_OUTPUT
+            echo "label=MI355" >> $GITHUB_OUTPUT
+          fi
+      - name: Run Container for ${{ matrix.jax-version }}
+        run: |
+          docker run -d \
+            --name te-build-jax${{ matrix.jax-version }} \
+            --network=host \
+            --device=/dev/kfd \
+            --device=/dev/dri \
+            --ipc=host \
+            --shm-size 32G \
+            --group-add video \
+            --cap-add=SYS_PTRACE \
+            --security-opt seccomp=unconfined \
+            -e NVTE_FRAMEWORK=${{ env.NVTE_FRAMEWORK }} \
+            -e NVTE_USE_ROCM=${{ env.NVTE_USE_ROCM }} \
+            -e HIP_PLATFORM=${{ env.HIP_PLATFORM }} \
+            -e NVTE_ROCM_ARCH=${{ env.ROCM_ARCH }} \
+            -e NVTE_FUSED_ATTN_AOTRITON=${{ env.NVTE_FUSED_ATTN_AOTRITON }} \
+            -e CU_NUM=${{ env.CU_NUM }} \
+            -e PIP_BREAK_SYSTEM_PACKAGES=1 \
+            -v "${{ github.workspace }}:/workspace" \
+            ${{ matrix.docker-image }} \
+            tail -f /dev/null
+      - name: Build TE-wheel
```

Comment:
The Run Container ..., Build TE-wheel, and Cleanup containers steps can all be combined into one step that uses `docker run --rm`. I'd very much prefer that since it's much easier to read and you don't have to worry about messing up the cleanup step and leaving a bunch of junk around on the runner's system. I know that the other perf workflow does it like that, and I'd wish we'd changed it before it was merged.